File: UI/angular11/src/app/dtos/order/order-create-dto.ts
import { CityReadDto } from "../city/city-read-dto"
import { UserReadDto } from "../user/user-read-dto"

export class OrderCreateDto {
    user: UserReadDto
    prePaid: number
    postPaid: number
    demands: string
    isRoot: boolean
    city: CityReadDto
    subOrders: OrderCreateDto[]
}

File: ConstructionQualityControl.Domain/Dtos/Order/OrderCreateDto.cs
using System.Collections.Generic;

namespace ConstructionQualityControl.Domain.Dtos
{
    public class OrderCreateDto
    {
        public UserReadDto User { get; set; }
        public decimal PrePaid { get; set; }
        public decimal PostPaid { get; set; }
        public string Demands { get; set; }
        public bool IsRoot { get; set; }
        public List<OrderCreateDto> SubOrders { get; set; }
    }
}
